lib/core/request.js

```diff
@@ -78,6 +79,10 @@ class Request {
       this.body = body.length ? body : null
     } else if (typeof body === 'string') {
       this.body = body.length ? Buffer.from(body) : null
+    } else if (util.isIterator(body)) { // check this after buffer and string so it doesn't return false positives
+      this.body = Readable.from(body)
+    } else if (util.isGenerator(body)) {
+      this.body = Readable.from(body())
```

Comment:
I think you might be able to skip the checks here and just pass it to `Readable.from` and catch INVALID_ARG_TYPE errors from there. That way you don't need to re-implement the type checks.

Reply:
Hmm okay I also found this line in the [documentaion](https://nodejs.org/dist/latest-v14.x/docs/api/stream.html#stream_stream_readable_from_iterable_options):

> Calling Readable.from(string) or Readable.from(buffer) will not have the strings or buffers be iterated to match the other streams semantics for performance reasons.

So like is it potentially safe to just always pass anything but an existing stream to Readable.from?